python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -162,4 +162,71 @@ def _isUsable(self, source):
             and not source.get(self.fluxFlagKey) \
             and self._goodSN(source)
 
+
+class MatcherPessimisticSourceSelectorTask(MatcherSourceSelectorTask):
```

Comment:
I'd rather leave this as MatcherPessimistic... as it is a subclass of the matcher source selector. This source selector is somewhat temporary as the hope is to replace the current matcher with matcherPessimisticB. matcherSourceSelector will then become equivalent to matcherPessimisticSourceSelector and the later will be removed. I'm keeping the old one around to preserve most of the current matcher behavior before it is replaced.